ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/QueryFilterNode.java

```diff
@@ -340,16 +340,10 @@ public Void visitComparisonExpression(
                 "Bound on non-existent column " + columnName, isWindowed));
 
         if (col.namespace() == Namespace.KEY) {
-          if (node.getType() != Type.EQUAL) {
-            setTableScanOrElseThrow(() ->
-                invalidWhereClauseException("Bound on key columns '"
-                        + getSource().getSchema().key() + "' must currently be '='", isWindowed));
-          }
           if (seenKeys.get(col.index())) {
             setTableScanOrElseThrow(() -> invalidWhereClauseException(
-                "An equality condition on the key column cannot be combined with other comparisons"
-                    + " such as an IN predicate",
-                isWindowed));
+                "A comparison condition on the key column cannot be combined with other"
```

Comment:
Regarding doing a table scan if there's more than one range query, just a suggestion about that check: 

In this validator, you probably want to keep track of `requiresRangeScan`, and then where the validator is called in `validateWhereClauseAndCheckTableScan`, you can count the number of them across all disjuncts and if you see more than one, return a table scan for the whole thing rather than doing multiple range scans.

Reply:
For queries with multiple range expressions we currently fall back to table scans. That code is validateWhereClause and was there even before this PR

Reply:
I'm not referring to more than one range in a single disjunct (e.g. `KEY < 'a' AND KEY > 'b'`), I'm referring to multiple disjuncts: `KEY < 'a' OR KEY > 'b'`.  The latter would do multiple scans and require deduping or can be done in a single table scan.  I'm hoping to do a table scan in the latter case too.

Reply:
Multiple range expressions, both across disjuncts or within a single disjuncts fall back to table scan. Added tests to cover those cases.